File: src/main/java/ba/aljovic/amer/movierecommendation/configuration/NaiveBayesConfig.java
package ba.aljovic.amer.movierecommendation.configuration;

import ba.aljovic.amer.movierecommendation.application.algorithm.NaiveBayesAlgorithm;
import ba.aljovic.amer.movierecommendation.application.algorithm.Recommendation;
import ba.aljovic.amer.movierecommendation.application.algorithm.RecommendationAlgorithm;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Profile;

@Configuration
@Profile("algorithm-naiveBayes")
public class NaiveBayesConfig
{
    // Naive Bayes
    @Bean
    public Recommendation naiveBayesRecommendation()
    {
        return new Recommendation.RecommendationBuilder(naiveBayesAlgorithm()).build();
    }

    @Bean
    public RecommendationAlgorithm naiveBayesAlgorithm()
    {
        return new NaiveBayesAlgorithm(1);
    }
}

File: src/main/java/ba/aljovic/amer/movierecommendation/configuration/NaiveBayesConfig.scala
package ba.aljovic.amer.movierecommendation.configuration

import ba.aljovic.amer.movierecommendation.application.algorithm.{NaiveBayesAlgorithm, Recommendation, RecommendationAlgorithm}
import org.springframework.context.annotation.{Bean, Configuration, Profile}

@Configuration
@Profile(Array("algorithm-naiveBayes"))
class NaiveBayesConfig
{
  @Bean
  def naiveBayesRecommendation: Recommendation = {
    Recommendation(naiveBayesAlgorithm)
  }

  @Bean
  def naiveBayesAlgorithm: RecommendationAlgorithm = new NaiveBayesAlgorithm(1, 5, 9)
}
